### PR TITLE
feat: Breadcrumb  support DropDown

### PR DESCRIPTION
**Pull request by chenshuai2144**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

https://github.com/ant-design/ant-design/issues/16287

### 💡 Solution

image

### 📝 Changelog

- Breadcrumb support overlay props.
- Breadcrumb 支持 overlay 属性

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/breadcrumb/demo/basic.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/breadcrumb/demo/basic.md)
[View rendered components/breadcrumb/demo/overlay.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/breadcrumb/demo/overlay.md)
[View rendered components/breadcrumb/demo/router.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/breadcrumb/demo/router.md)
[View rendered components/breadcrumb/index.en-US.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/breadcrumb/index.en-US.md)
[View rendered components/breadcrumb/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/breadcrumb/index.zh-CN.md)
[View rendered components/button/demo/basic.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/demo/basic.md)
[View rendered components/button/demo/block.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/demo/block.md)
[View rendered components/button/demo/disabled.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/demo/disabled.md)
[View rendered components/button/demo/ghost.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/demo/ghost.md)
[View rendered components/button/demo/size.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/demo/size.md)
[View rendered components/button/index.en-US.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/index.en-US.md)
[View rendered components/button/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/breadcrumb-menu/components/button/index.zh-CN.md)

### CODE REVIEW

File: components/breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -69,7 +72,44 @@ export default class Breadcrumb extends React.Component<BreadcrumbProps, any> {
...
+  genForRoutes = ({
+    routes = [],
+    params = {},
+    separator,
+    itemRender = defaultItemRender,
+  }: BreadcrumbProps) => {
+    const paths: string[] = [];
+    return routes.map(route => {
+      route.path = route.path || '';
+      let path: string = route.path.replace(/^\//, '');
```
Comment (ztplz) [inline components/breadcrumb/Breadcrumb.tsx]: 有必要再声明string吗  它返回的好像就是string

**Review by ztplz — DISMISSED** [review]

File: components/breadcrumb/demo/overlay.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+---
+order: 0
```
Comment (zombieJ) [inline components/breadcrumb/demo/overlay.md]: order 不对

File: components/breadcrumb/demo/overlay.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+title:
+  zh-CN: 基本
+  en-US: Basic Usage
```
Comment (zombieJ) [inline components/breadcrumb/demo/overlay.md]: 标题不对

File: components/breadcrumb/BreadcrumbItem.tsx [hunk 4]
```diff
@@ -50,6 +62,25 @@ export default class BreadcrumbItem extends React.Component<BreadcrumbItemProps,
...
+   * Wrap a DropDown
+   */
+  wrapToDropDown = (breadcrumbItem: React.ReactNode) => {
+    const { overlay } = this.props;
+    if (overlay) {
+      return (
+        <DropDown overlay={overlay} placement="bottomCenter">
+          <a className="ant-dropdown-link">
+            {breadcrumbItem}
+            &nbsp;
```
Comment (zombieJ) [inline components/breadcrumb/BreadcrumbItem.tsx]: = =！略丑，改成 `{' '}` 假装看不见吧。

File: components/breadcrumb/Breadcrumb.tsx (hunk 1 above)
Comment (chenshuai2144) [reply components/breadcrumb/Breadcrumb.tsx]: 顺手了. fixed

File: components/breadcrumb/BreadcrumbItem.tsx [hunk 5]
```diff
@@ -50,6 +62,25 @@ export default class BreadcrumbItem extends React.Component<BreadcrumbItemProps,
     return null;
   };
 
+  /**
+   * if overlay is have
+   * Wrap a DropDown
+   */
+  wrapToDropDown = (breadcrumbItem: React.ReactNode) => {
```
Comment (zombieJ) [inline components/breadcrumb/BreadcrumbItem.tsx]: `wrapToDropDown` => `renderBreadcrumbNode`

File: components/breadcrumb/BreadcrumbItem.tsx (hunk 4 above)
Comment (chenshuai2144) [reply components/breadcrumb/BreadcrumbItem.tsx]: {' '} 感觉更丑

File: components/breadcrumb/BreadcrumbItem.tsx (hunk 5 above)
Comment (chenshuai2144) [reply components/breadcrumb/BreadcrumbItem.tsx]: 如果是 renderBreadcrumbNode 我觉得可以拆出去一个给用户.
所以用了 wrapToDropDown

File: components/breadcrumb/BreadcrumbItem.tsx (hunk 4 above)
Comment (zombieJ) [reply components/breadcrumb/BreadcrumbItem.tsx]: 有道理，那改成样式处理吧……

File: components/breadcrumb/BreadcrumbItem.tsx (hunk 5 above)
Comment (zombieJ) [reply components/breadcrumb/BreadcrumbItem.tsx]: 不用给用户，我们内部渲染子组件代码都是 `renderXXX` 命名的。

File: components/breadcrumb/demo/overlay.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+);
+
+ReactDOM.render(
+  <div>
+    <Breadcrumb>
+      <Breadcrumb.Item>Ant Design</Breadcrumb.Item>
+      <Breadcrumb.Item>
+        <a href="">Component</a>
+      </Breadcrumb.Item>
+      <Breadcrumb.Item overlay={menu}>
```
Comment (afc163) [inline components/breadcrumb/demo/overlay.md]: overlay 改成 menu 好了。

Comment (chenshuai2144) [reply components/breadcrumb/demo/overlay.md]: https://ant.design/components/dropdown-cn/
这里叫 overlay 应该保持统一吧

Comment (afc163) [reply components/breadcrumb/demo/overlay.md]: 👌

**Review by zombieJ — APPROVED** [review]

**Pull request merged**